File: competition/lc/2.py
"""
You are given two non-empty linked lists representing two non-negative integers. The digits are stored in reverse order, and each of their nodes contains a single digit. Add the two numbers and return the sum as a linked list.

You may assume the two numbers do not contain any leading zero, except the number 0 itself.

M: 2 → 4 → 3   = 342
N: 5 → 6 → 4   = 465

342 + 465 = 807 → output: 7 → 0 → 8
"""

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def traverse(L: ListNode):
    pass
def addTwoNumbers(l1: ListNode, l2: ListNode):
    pass


M_0 = ListNode(2)
M_1 = ListNode(4)
M_2 = ListNode(3)

N_0 = ListNode(5)
N_1 = ListNode(6)
N_2 = ListNode(4)


M_0.next = M_1
M_1.next = M_2

N_0.next = N_1
N_1.next = N_2


if __name__ == "__main__":
    demo = addTwoNumbers(M_0, N_0)
    traverse(demo)

File: competition/leetcode/2.cpp
#include <bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode* next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

void traverse(ListNode* L) {

}

void deleteList(ListNode* head) {
    while (head != nullptr) {
        ListNode* temp = head;
        head = head->next;
        delete temp;
    }
}

ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {


    return new ListNode();
}

int main() {
// M list: 2 -> 4 -> 3
    ListNode* M_0 = new ListNode(2);
    ListNode* M_1 = new ListNode(4);
    ListNode* M_2 = new ListNode(3);

    M_0->next = M_1;
    M_1->next = M_2;

    // N list: 5 -> 6 -> 4
    ListNode* N_0 = new ListNode(5);
    ListNode* N_1 = new ListNode(6);
    ListNode* N_2 = new ListNode(4);

    N_0->next = N_1;
    N_1->next = N_2;

    ListNode* demo = addTwoNumbers(M_0, N_0);
    traverse(demo);

    deleteList(demo);
    deleteList(M_0);
    deleteList(N_0);

    return 0;
}













